selene-vscode/src/util.ts

```diff
@@ -84,13 +82,17 @@ function getSeleneFilenamePattern(): RegExp {
     }
 }
 
-async function fileExists(filename: string): Promise<boolean> {
-    return fs.promises.stat(filename)
-        .then(() => true)
-        .catch((error) => error.code !== "ENOENT")
+async function fileExists(filename: vscode.Uri): Promise<boolean> {
+    try {
+        await vscode.workspace.fs.stat(filename);
+        return true;
+    } catch (err) {
+        // If an error was thrown, the file was not found
+        return false;
+    }

```

Comment:
Why change this from what there was before? `Promise.catch` should work perfectly fine.

Reply:
Plus, what was there before captures in the case of *other* failures that can happen during I/O. I'd be surprised if vscode.workspace.fs.stat didn't provide something similar, if not identical to `error.code`.

Reply:
`vscode.workspace.fs.stat(filename)` returns a `Thenable`, so it only has a `.then()`, but no `.catch()` - it will just throw the error.
For your second point, I could change it to
```suggestion
    try {
        await vscode.workspace.fs.stat(filename);
        return true;
    } catch (err) {
        return err != vscode.FileSystemError.FileNotFound;
    }
```

Reply:
Ugh, that's annoying. Yes, that's better than a blanket catch.